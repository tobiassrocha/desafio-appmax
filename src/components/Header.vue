<template>
    <header class="header">
      <div class="header__container">
        <a href="/">
          <h1 class="header__logo"></h1>
        </a>
        <nav class="navbar">
          <ul class="navbar__list">
            <li class="navbar__item">
              <a href="/" class="navbar__link">Home</a>
            </li>
            <li class="navbar__item">
              <a href="/create-form-page" class="navbar__link">Criar nova oferta</a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
</template>

<style lang="less" escoped>
.header {
  height: 95px;
  
  &__container {
    background-color: #fff;
    border-bottom: 1px solid #000;
    box-sizing: border-box;
    display: flex;
    justify-content: space-between;
    padding: 10px 30px;
    position: fixed;
    width: 100%;
  }
  &__logo {
    background-image: url('../assets/logo.png');
    background-size: 100%;
    background-repeat: no-repeat;
    margin: 0;
    width: 300px;
    height: 75px;
  }

  .navbar {
    display: flex;

    &__list {
      display: flex;
      list-style: none;
      margin: 0;
      padding: 0;
      align-items: center;
    }
    
    &__item {
      font-size: 16px;
      text-decoration: none;

      &:first-child {
        margin-right: 15px;
      }
    }
    
    &__link {
      color: #000;
      font-size: 16px;
      font-weight: bold;
      text-decoration: none;
      border-radius: 5px;
      border: 1px solid;
      padding: 5px 15px;

      &:hover {
        background-color: #fcbf10;
      }
    }
  }

  @media (max-width: 768px) {

    height: 75px;

    .navbar {
      display: none;
    }

    &__container {
      justify-content: center;
    }

    &__logo {
      width: 230px;
      height: 55px;
    }
  }
}
</style>