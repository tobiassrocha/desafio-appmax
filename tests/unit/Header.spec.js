import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Header'

describe('Header', () => {
  it('renders the component', () => {
    // arrange
    const wrapper = shallowMount(Header)

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})