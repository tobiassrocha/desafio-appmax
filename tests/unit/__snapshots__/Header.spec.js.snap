// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the component 1`] = `
<header class="header">
  <div class="header__container"><a href="/">
      <h1 class="header__logo"></h1>
    </a>
    <nav class="navbar">
      <ul class="navbar__list">
        <li class="navbar__item"><a href="/" class="navbar__link">Home</a></li>
        <li class="navbar__item"><a href="/create-form-page" class="navbar__link">Criar nova oferta</a></li>
      </ul>
    </nav>
  </div>
</header>
`;
